backend/connector-integration/src/connectors/getnet/transformers.rs

```
@@ -0,0 +1,692 @@
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+
+    fn try_from(
+        wrapper: GetnetRouterData<
+            RouterDataV2<
+                Authorize,
+                PaymentFlowData,
+                PaymentsAuthorizeData<T>,
+                PaymentsResponseData,
+            >,
+            T,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let item = &wrapper.router_data;
+        let card_data = match &item.request.payment_method_data {
+            PaymentMethodData::Card(card) => card,
+            _ => {
+                return Err(errors::ConnectorError::NotImplemented(
+                    "Payment method not supported".to_string(),
+                )
+                .into())
+            }
+        };
+
+        // Convert 4-digit year to 2-digit year (e.g., "2025" -> "25")
+        let year_str = card_data.card_exp_year.peek();
+        let expiration_year = if year_str.len() >= 2 {
+            Secret::new(year_str[year_str.len() - 2..].to_string())
+        } else {
+            card_data.card_exp_year.clone()
+        };
+
+        let cardholder_name = if let Some(ref holder_name) = card_data.card_holder_name {
+            holder_name.clone()
+        } else if let Some(billing_addr) = item.resource_common_data.get_optional_billing() {
+            let first_name = billing_addr
+                .address
+                .as_ref()
+                .and_then(|addr| addr.first_name.as_ref().map(|n| n.peek().clone()))
+                .unwrap_or_default();
+            let last_name = billing_addr
+                .address
+                .as_ref()
+                .and_then(|addr| addr.last_name.as_ref().map(|n| n.peek().clone()))
+                .unwrap_or_default();
+
+            if !first_name.is_empty() || !last_name.is_empty() {
+                Secret::new(format!("{} {}", first_name, last_name).trim().to_string())
+            } else {
+                Secret::new(DEFAULT_CARD_HOLDER_NAME.to_string())
```

Throw a BadRequest if no card holder is there rather than hardcoding

---

Done